File: firmware/src/provider_visuals.cpp
#include "provider_visuals.h"

#include <cstring>

namespace agentmeter {

ProviderVisuals provider_visuals(const char* provider_id) {
  if (std::strcmp(provider_id, "codex") == 0) {
    return ProviderVisuals{0x3DDC97, ProviderMark::Codex};
  }
  if (std::strcmp(provider_id, "claude") == 0) {
    return ProviderVisuals{0xF4A261, ProviderMark::Claude};
  }
  if (std::strcmp(provider_id, "gemini") == 0) {
    return ProviderVisuals{0x6FA8FF, ProviderMark::Gemini};
  }
  if (std::strcmp(provider_id, "cursor") == 0) {
    return ProviderVisuals{0xD8D8DC, ProviderMark::CursorCube};
  }
  return ProviderVisuals{0x8B7CFF, ProviderMark::Initial};
}

}  // namespace agentmeter

File: desktop/Sources/AgentMeterCore/ProviderVisuals.swift
public enum ProviderVisuals {
    public static func accentHex(for providerId: String) -> UInt32 {
        switch providerId.lowercased() {
        case "codex": 0x3DDC97
        case "claude": 0xF4A261
        case "gemini": 0x6FA8FF
        case "cursor": 0xD8D8DC
        default: 0x8B7CFF
        }
    }
}
